src/backends/all.rs

```diff
@@ -0,0 +1,171 @@
+use std::collections::{BTreeMap, BTreeSet};
+
+use crate::prelude::*;
+use anyhow::Result;
+
+macro_rules! append {
+    ($($backend:ident),*) => {
+        pub fn append(&mut self, other: &mut Self) {
+            $(
+                self.$backend.append(&mut other.$backend);
+            )*
+        }
+    };
+}
+macro_rules! is_empty {
+    ($($backend:ident),*) => {
+        pub fn is_empty(&self) ->bool {
+            $(
+                self.$backend.is_empty() &&
+            )* true
+        }
+    };
+}
+macro_rules! to_package_ids {
+    ($($backend:ident),*) => {
+        pub fn to_package_ids(&self) -> PackageIds {
+            PackageIds {
+                $(
+                    $backend: self.$backend.keys().cloned().collect(),
+                )*
+            }
+        }
+    };
+}
+
+macro_rules! x {
+    ($($backend:ident),*) => {
+        #[derive(Debug, Clone, Default)]
+        #[allow(non_snake_case)]
+        pub struct PackageIds {
+            $(
+                pub $backend: BTreeSet<<$backend as Backend>::PackageId>,
+            )*
+        }
+        impl PackageIds {
```

Comment:
Should probably be split into multiple macros. I don't understand having a single large macro for this purpose.

Reply:
See above, but I wouldn't mind splitting it